include/GafferScene/SceneAlgo.h

```diff
@@ -73,6 +74,19 @@ void matchingPaths( const Filter *filter, const ScenePlug *scene, PathMatcher &p
 /// FilteredSceneProcessor::filterPlug() would be passed.
 void matchingPaths( const Gaffer::IntPlug *filterPlug, const ScenePlug *scene, PathMatcher &paths );
 
+/// Calls a functor on all paths in the scene
+/// The functor must take ( const ScenePlug*, const ScenePlug::ScenePath& ), and can return false to prune traversal
+template <class ThreadableFunctor>
+void parallelTraverse( const ScenePlug *scene, ThreadableFunctor &f, const Gaffer::Context *context = Gaffer::Context::current() );
```

Comment:
I may have confused the issue in my earlier review comments, but I don't think the context should be an argument to `parallelTraverse()`. Everywhere (almost - there are a couple of places to be fixed) in the API, the context in which an action is performed is implicit, and is specified by making a context current before calling a method.

We do sometimes pass a context explicitly in the internals of the implementation of an action, to say "here is the current context, this should save you the time you'd otherwise take to retrieve it with `Context::current()`". So I was suggesting that we might want to have the context be an argument to the Functor for that reason. I'm open to discussion on that - it seems to be an awkward middle ground between external and internal code.